src/main/webapp/app/exercises/shared/participation/participation.service.ts

```diff
@@ -29,39 +28,33 @@ export class ParticipationService {
         const copy = this.convertDateFromClient(participation);
         return this.http
             .put<StudentParticipation>(SERVER_API_URL + `api/exercises/${exerciseId}/participations`, copy, { observe: 'response' })
-            .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)))
-            .pipe(map((res: EntityResponseType) => this.adjustRepositoryUrlFromServer(res)));
+            .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
     }
 
     find(participationId: number): Observable<EntityResponseType> {
         return this.http
             .get<StudentParticipation>(`${this.resourceUrl}/${participationId}`, { observe: 'response' })
-            .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)))
-            .pipe(map((res: EntityResponseType) => this.adjustRepositoryUrlFromServer(res)));
+            .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
     }
 
     findWithLatestResult(participationId: number): Observable<EntityResponseType> {
         return this.http
             .get<StudentParticipation>(`${this.resourceUrl}/${participationId}/withLatestResult`, { observe: 'response' })
-            .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)))
-            .pipe(map((res: EntityResponseType) => this.adjustRepositoryUrlFromServer(res)));
+            .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
     }
 
     /*
      * Finds one participation for the currently logged in user for the given exercise in the given course
      */
     findParticipation(exerciseId: number): Observable<EntityResponseType | null> {
-        return this.http
-            .get<StudentParticipation>(SERVER_API_URL + `api/exercises/${exerciseId}/participation`, { observe: 'response' })
-            .pipe(
-                map((res: EntityResponseType) => {
-                    if (res == undefined) {
-                        return null;
-                    }
-                    return this.convertDateFromServer(res);
-                }),
-            )
-            .pipe(map((res: EntityResponseType) => this.adjustRepositoryUrlFromServer(res)));
+        return this.http.get<StudentParticipation>(SERVER_API_URL + `api/exercises/${exerciseId}/participation`, { observe: 'response' }).pipe(
+            map((res: EntityResponseType) => {
+                if (res == undefined) {
+                    return null;
+                }
+                return this.convertDateFromServer(res);
+            }),
```

Comment:
My suggestion:
```suggestion
            map((res: EntityResponseType) => res ? this.convertDateFromServer(res) : null),
```

If you want to leave it in this structure, then at least:
```suggestion
            map((res: EntityResponseType) => {
                if (res === undefined) {
                    return null;
                }
                return this.convertDateFromServer(res);
            }),
```

Reply:
== is better here, because it covers null and undefined
=== would only cover undefined, null would get through

Reply:
Then we should just use ``!res`` instead or the first version I posted

Reply:
I looked into it, I don't think that the response can even be null or undefined. If at all, the body will be null.